zendesk/ticket.go

```diff
@@ -0,0 +1,82 @@
+package zendesk
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"time"
+)
+
+type Ticket struct {
+	ID              int64         `json:"id,omitempty"`
+	URL             string        `json:"url,omitempty"`
+	ExternalID      string        `json:"external_id,omitempty"`
+	Type            string        `json:"type,omitempty"`
+	Subject         string        `json:"subject,omitempty"`
+	RawSubject      string        `json:"raw_subject,omitempty"`
+	Description     string        `json:"description,omitempty"`
+	Priority        string        `json:"priority,omitempty"`
+	Status          string        `json:"status,omitempty"`
+	Recipient       string        `json:"recipient,omitempty"`
+	RequesterID     int64         `json:"requester_id"`
+	SubmitterID     int64         `json:"submitter_id,omitempty"`
+	AssigneeID      int64         `json:"assignee_id,omitempty"`
+	OrganizationID  int64         `json:"organization_id,omitempty"`
+	GroupID         int64         `json:"group_id,omitempty"`
+	CollaboratorIDs []int64       `json:"collaborator_ids,omitempty"`
+	Collaborators   Collaborators `json:"collaborators,omitempty"`
```

Comment:
it might be good to have one of these setup in the json fixture so that we can verify that this is set properly by the client

Reply:
I found the description about `collaborators` field in Zendesk API documentation. It says

> POST requests only. Users to add as cc's when creating a ticket.

This seems to be a virtual field to add collaborator. Internally, this payload would be converted to actual Zendesk user record and its ID would be added to `collaborator_ids`.

So, I added only value of `collaborator_ids` because `collaborators` key will never appear in GET response. (86af6bc)

```diff
diff --git a/fixture/GET/tickets.json b/fixture/GET/tickets.json
index 132fe28..911beb7 100644
--- a/fixture/GET/tickets.json
+++ b/fixture/GET/tickets.json
@@ -19,7 +19,7 @@
         }
       },
       "created_at": "2019-06-03T02:23:47Z",
-      "updated_at": "2019-06-03T02:25:28Z",
+      "updated_at": "2019-06-05T01:13:24Z",
       "type": null,
       "subject": "Mail to create fixture ticket for testing",
       "raw_subject": "Mail to create fixture ticket for testing",
@@ -32,8 +32,12 @@
       "assignee_id": 377922500012,
       "organization_id": 360363695492,
       "group_id": 360004077472,
-      "collaborator_ids": [],
-      "follower_ids": [],
+      "collaborator_ids": [
+        377922500012
+      ],
+      "follower_ids": [
+        377922500012
+      ],
       "email_cc_ids": [],
       "forum_topic_id": null,
       "problem_id": null,
```

FYI: Also, `follower_ids` automatically added by internal logic.

Reply:
OK we can worry about that then when we get to implementing the POST